core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -204,13 +204,21 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
           mod1 >= mod2
       }
 
+      val tasks = mutable.ListBuffer.empty[Future[_]]
       logInfos.sliding(20, 20).foreach { batch =>
```

Comment:
Thanks! I am a Scala noob - I did it the Java way. :-)
Your suggestion looks much cleaner - I have updated the PR.